docs/designers-developers/developers/filters/block-filters.md

```diff
@@ -237,8 +237,9 @@ Adding blocks is easy enough, removing them is as easy. Plugin or theme authors
 
 ```js
 // my-plugin.js
-
-wp.blocks.unregisterBlockType( 'core/verse' );
+wp.domReady( function() {
```

Comment:
```suggestion
window._wpLoadGutenbergEditor.then( function() {
  wp.blocks.unregisterBlockType( "core/verse" );
});
```

Reply:
I think `_wpLoadGutenbergEditor` starts with underscore to state explicitely that it's private and shouldn't be used.

Reply:
@youknowriad I've summarized my thoughts [here](https://github.com/WordPress/gutenberg/pull/12613#issuecomment-447058893). I'd love using a better alternative, but to my knowledge, we don't have one?

Reply:
domReady don't work?

Reply:
At that point, core blocks aren't yet registered, so they can't be unregistered. It's a race condition because we initialize the editor precisely using [domReady](https://github.com/WordPress/gutenberg/blob/8790bce7ae56191d00fe7384ea1832df10624961/lib/client-assets.php#L1335).

Reply:
It's too soon : on DomReady Gutenberg hasn't yet registered blocks. 

So we should do it later. 

But in the future, maybe UnregisterBlockType should work differently :  instead of deregistering blocks, it should prevent its initialization. It would make more sense : instead of waiting for the right time, and deregister a just registered block , it should just say, from the beginning : "you wont be initialized at all".

Reply:
https://github.com/WordPress/gutenberg/blob/8790bce7ae56191d00fe7384ea1832df10624961/lib/client-assets.php#L1337-L1341

Gutenberg does register the blocks on dom ready, just make sure it's run after that on the JS loop. (could by adding `edit-post` as a dependency and/or `setTimeout(callback)`

Reply:
Just to clarify neither of these solutions is perfect, but that's what we have at the moment.

Reply:
I've found an alternative to using `_wpLoadGutenbergEditor`. So, this code:

```js
window.wp.domReady(function() {
  wp.blocks.unregisterBlockType("core/verse");
});
```

* **doesn't** work if we declare `wp-blocks` and `wp-dom-ready` as dependencies.
* **does** work if we declare `wp-blocks`, `wp-dom-ready`, and `wp-edit-post` as dependencies.

So, my take is that if we want to execute anything _after the editor has been initialized_, it suffices with listing `edit-post` as a dependency. 

Not sure which one I prefer until we develop a better mechanism. I think I have a slight preference to use `_wpLoadGutenbergEditor` because is more obvious.

Reply:
GitHub is playing out for me and just showed me @youknowriad comment above where he suggests adding `wp-edit-post` as a dependency also works. I'd go with whatever preference the Gutenberg core team has for this.

Reply:
Yes, this code and the `wp-edit-post` extra dependency sound good to me.
```
window.wp.domReady(function() {
  wp.blocks.unregisterBlockType("core/verse");
});
```

Reply:
Just tested it and it works ! (and it's elegant) 

Thanks !

Reply:
Oh no, it's more complex than that. I ran some tests with my plugin and here is what I've found : 

1/ with the dirty way (_wpLoadBlockEditor), if I disable an heading and go back on a post with headings, I get this : 

![capture 2018-12-14 a 10 30 42](https://user-images.githubusercontent.com/1785489/49995024-5405bf80-ff8b-11e8-9a41-06fe95b601eb.jpg)

It's the way I expect things to work

2/ But, if I use the cleaner way (window.wp.domReady), I get an editor crash: 

![capture 2018-12-14 a 10 29 03](https://user-images.githubusercontent.com/1785489/49995070-6da70700-ff8b-11e8-9f98-7b2a7f932794.jpg)

(don't mind the first error, it's from woo api because i wasn't in https)

Isn't that crazy?